django/contrib/postgres/forms/jsonb.py

```diff
@@ -22,15 +26,23 @@ def to_python(self, value):
             return value
         if value in self.empty_values:
             return None
+        elif isinstance(value, (list, dict, int, float, bool, JsonStr)):
+            return value
+
         try:
-            return json.loads(value)
+            converted = json.loads(value)
         except ValueError:
             raise forms.ValidationError(
                 self.error_messages['invalid'],
                 code='invalid',
                 params={'value': value},
             )
 
+        if isinstance(converted, six.text_type):
+            return JsonStr(converted)
```

Comment:
@mjtamlyn can you take a look if returning a subclass of `six.text_type` could potentially break things somewhere down the line?